[2.0] Make mlflow.sagemaker.deploy() / delete() APIs internal, remove CLIs (#6650)
mlflow/mlflow: docs/source/models.rst

Deploy a ``python_function`` model on Amazon SageMaker
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> The :py:mod:`mlflow.deployments` and py:mod:`mlflow.sagemaker` modules can deploy

Reviewer: ```suggestion
The :py:mod:`mlflow.deployments` and :py:mod:`mlflow.sagemaker` modules can deploy
```

Author: Thanks for catching this!

Commands
~~~~~~~~~

* :py:func:`run-local <mlflow.sagemaker.run_local>` deploys the model locally in a Docker container.
  The image and the environment should be identical to how the model would be run remotely and it is
  therefore useful for testing the model prior to deployment.

* `build-and-push-container <cli.html#mlflow-sagemaker-build-and-push-container>`_ builds an MLfLow
  Docker image and uploads it to ECR. The caller must have the correct permissions set up. The image
  is built locally and requires Docker to be present on the machine that performs this step.

> * :py:func:`deploy <mlflow.sagemaker.SageMakerDeploymentClient>` deploys the model on Amazon

Reviewer: Is this line correct?

```suggestion
* :py:func:`create <mlflow.sagemaker.SageMakerDeploymentClient.creaet_deployment>` deploys the model on Amazon
```

Author: Fixed and made the link read `mlflow deployments create -t sagemaker`, also updated `run-local` to `mlflow deployments run-local -t sagemaker`

  SageMaker. MLflow uploads the Python Function model into S3 and starts an Amazon SageMaker
  endpoint serving the model.

.. rubric:: Example workflow using the MLflow CLI

.. code-block:: bash

    mlflow sagemaker build-and-push-container  - build the container (only needs to be called once)

>     mlflow deployments run-local -t sagemaker -m <path-to-model>  - test the model locally
>     mlflow deployments -t sagemaker create - deploy the model remotely

Reviewer: ```suggestion
    mlflow deployments run-local -t sagemaker -m <path-to-model> # test the model locally
    mlflow deployments -t sagemaker create # deploy the model remotely
```

Can we replace `-` with `#`? It looks like `- test the model locally` is a part of the command, but it's not.

Author: Absolutely! Fixed!

>     mlflow deployments -t sagemaker create - deploy the model remotely

Reviewer: ```suggestion
    mlflow deployments create -t sagemaker - deploy the model remotely
```

`mlflow deployments -t sagemaker create` gave:

```
mlflow deployments -t sagemaker create
Usage: mlflow deployments [OPTIONS] COMMAND [ARGS]...
Try 'mlflow deployments --help' for help.

Error: No such option: -t
```

Author: Great catch! Fixed!
